core/search/sort.hpp

```diff
@@ -361,13 +406,150 @@ struct score_traits {
     return const_cast<score_type&>(score_cast(const_cast<const byte_type*>(buf)));
   }
 
-  static void merge(const order_bucket* ctx, byte_type* dst,
-                    const byte_type** src_start, const size_t size) {
+  static void aggregate(const order_bucket* ctx, byte_type* dst,
+                        const byte_type** src_start, const size_t size) {
+    const auto offset = ctx->score_offset;
+    auto& casted_dst = score_cast(dst + offset);
+
+    switch (size) {
+      case 0:
+        casted_dst = ScoreType();
+        break;
+      case 1:
+        casted_dst = score_cast(src_start[0] + offset);
+        break;
+      case 2:
+        casted_dst = score_cast(src_start[0] + offset)
+                   + score_cast(src_start[1] + offset);
+        break;
+      case 3:
+        casted_dst = score_cast(src_start[0] + offset)
+                   + score_cast(src_start[1] + offset)
+                   + score_cast(src_start[2] + offset);
+        break;
+      case 4:
+        casted_dst = score_cast(src_start[0] + offset)
+                   + score_cast(src_start[1] + offset)
+                   + score_cast(src_start[2] + offset)
+                   + score_cast(src_start[3] + offset);
+        break;
+      case 5:
+        casted_dst = score_cast(src_start[0] + offset)
+                   + score_cast(src_start[1] + offset)
+                   + score_cast(src_start[2] + offset)
+                   + score_cast(src_start[3] + offset)
+                   + score_cast(src_start[4] + offset);
+        break;
+      case 6:
+        casted_dst = score_cast(src_start[0] + offset)
+                   + score_cast(src_start[1] + offset)
+                   + score_cast(src_start[2] + offset)
+                   + score_cast(src_start[3] + offset)
+                   + score_cast(src_start[4] + offset)
+                   + score_cast(src_start[5] + offset);
+        break;
+      case 7:
+        casted_dst = score_cast(src_start[0] + offset)
+                   + score_cast(src_start[1] + offset)
+                   + score_cast(src_start[2] + offset)
+                   + score_cast(src_start[3] + offset)
+                   + score_cast(src_start[4] + offset)
+                   + score_cast(src_start[5] + offset)
+                   + score_cast(src_start[6] + offset);
+        break;
+      case 8:
+        casted_dst = score_cast(src_start[0] + offset)
+                   + score_cast(src_start[1] + offset)
+                   + score_cast(src_start[2] + offset)
+                   + score_cast(src_start[3] + offset)
+                   + score_cast(src_start[4] + offset)
+                   + score_cast(src_start[5] + offset)
+                   + score_cast(src_start[6] + offset)
+                   + score_cast(src_start[7] + offset);
+        break;
+      default:
+        casted_dst = score_cast(src_start[0] + offset);
+        for (size_t i = 1; i < size; ++i) {
```

Comment:
Is std::accumulate() slower?

Reply:
I would need to write custom iterator to use `std::max_element` since I accumulate scores at a certain offset